spec/integration_spec.rb

```diff
@@ -366,6 +380,74 @@ def public?
   end
 end
 
+describe 'EnableCheckSettingsSynchronously' do
+  describe 'has settings changes' do
+    before(:each) do
+      allow(EnableCheckSettingsSynchronously).to receive(:algoliasearch_settings_changed?).and_return(true)
+    end
+
+    it 'should call set_setting!' do
+      expect_any_instance_of(Algolia::Search::Index).to receive(:set_settings!)
+      EnableCheckSettingsSynchronously.send(:algolia_ensure_init)
+    end
+
+    it 'should not call set_setting' do
+      expect_any_instance_of(Algolia::Search::Index).not_to receive(:set_settings)
+      EnableCheckSettingsSynchronously.send(:algolia_ensure_init)
+    end
+  end
+
+  describe 'has no settings changes' do
+    before(:each) do
+      allow(EnableCheckSettingsSynchronously).to receive(:algoliasearch_settings_changed?).and_return(false)
+    end
+
+    it 'should not call set_setting!' do
+      expect_any_instance_of(Algolia::Search::Index).not_to receive(:set_settings!)
+      EnableCheckSettingsSynchronously.send(:algolia_ensure_init)
+    end
+
+    it 'should not call set_setting' do
+      expect_any_instance_of(Algolia::Search::Index).not_to receive(:set_settings)
+      EnableCheckSettingsSynchronously.send(:algolia_ensure_init)
+    end
+  end
+end
+
+describe 'EnableCheckSettingsAsynchronously' do
+  describe 'has settings changes' do
+    before(:each) do
+      allow(EnableCheckSettingsAsynchronously).to receive(:algoliasearch_settings_changed?).and_return(true)
+    end
+
+    it 'should call set_setting' do
+      expect_any_instance_of(Algolia::Search::Index).to receive(:set_settings)
+      EnableCheckSettingsAsynchronously.send(:algolia_ensure_init)
+    end
+
+    it 'should not call set_setting!' do
+      expect_any_instance_of(Algolia::Search::Index).not_to receive(:set_settings)
```

Comment:
Shouldn't this be as follows?
```suggestion
      expect_any_instance_of(Algolia::Search::Index).not_to receive(:set_settings!)
```

Reply:
I think your suggestion is perfectly right.

On the other hand, I was wondering why this test passed.
After additional investigation, it turns out that I have overlooked the cache mechanism.
When `algolia_ensure_init` is called more than once, neither `set_settings` nor `set_settings!` were being called ever at the second time and after, under any conditions.
I should have used the Red-Green-Refactor approach, but I skipped the Red check phase. embarrassing.

I have fixed the test, please check it out.